eo-maven-plugin/src/main/java/org/eolang/maven/Dependencies.java

```diff
@@ -0,0 +1,178 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.jcabi.log.Logger;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import javax.json.JsonValue;
+import org.apache.maven.model.Dependency;
+
+/**
+ * Maven dependencies abstraction.
+ *
+ * @since 0.28.11
+ */
+interface Dependencies {
+
+    /**
+     * Converts to a plain list.
+     *
+     * @return List of Maven Dependencies
+     */
+    List<Dependency> toList();
+
+    /**
+     * Filtered dependencies.
+     *
+     * @since 0.28.11
+     */
+    class FilteredDependencies implements Dependencies {
+
+        /**
+         * Decorated.
+         */
+        private final Dependencies dependencies;
+
+        /**
+         * All filters.
+         */
+        private final Collection<Predicate<Dependency>> filters;
+
+        /**
+         * The main constructor.
+         *
+         * @param dependencies Decorated
+         * @param filters All fiters
+         */
+        FilteredDependencies(
+            final Dependencies dependencies,
+            final Collection<Predicate<Dependency>> filters
+        ) {
+            this.dependencies = dependencies;
+            this.filters = filters;
+        }
+
+        @Override
+        public List<Dependency> toList() {
+            return this.dependencies.toList()
+                .stream()
+                .filter(this::applyAllFilters)
+                .collect(Collectors.toList());
+        }
+
+        /**
+         * Apply all filters for Dependency.
+         *
+         * @param dependency Dependency
+         * @return True if all filters were passed
+         */
+        private boolean applyAllFilters(final Dependency dependency) {
+            return this.filters.stream().allMatch(f -> f.test(dependency));
+        }
+    }
+
+    /**
+     * Dependencies uploaded from json file.
+     *
+     * @since 0.28.11
+     */
+    class JsonDependencies implements Dependencies {
+
+        /**
+         * File path.
+         */
+        private final Path file;
+
+        /**
+         * The main constructor.
+         *
+         * @param file File path
+         */
+        JsonDependencies(final Path file) {
+            this.file = file;
+        }
+
+        @Override
+        public List<Dependency> toList() {
+            try {
+                final List<Dependency> all = new ArrayList<>(0);
+                if (Files.exists(this.file)) {
+                    Logger.info(this, String.format("Dependencies file: %s", this.file));
```

Comment:
@volodya-lombrozo Can we set this logger level to `debug`? To avoid output noise

Reply:
@Graur Fixed. Please, take a look.